pkg/agent/nodeportlocal/k8s/handlers.go

```diff
@@ -0,0 +1,128 @@
+// +build !windows
+
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+)
+
+func (c *Controller) addRuleForPod(pod *corev1.Pod) error {
+	podIP, nodeIP := pod.Status.PodIP, pod.Status.HostIP
+	if podIP == "" || nodeIP == "" {
+		return nil
+	}
+	podContainers := pod.Spec.Containers
+
+	for _, container := range podContainers {
+		for _, cport := range container.Ports {
+			port := int(cport.ContainerPort)
+			if c.portTable.RuleExists(podIP, port) {
+				continue
+			}
+			nodePort, err := c.portTable.AddRule(podIP, port)
+			if err != nil {
+				return err
+			}
+			assignPodAnnotation(pod, nodeIP, port, nodePort)
+		}
+	}
+	return nil
+}
+
+// HandleDeletePod handles pod annotations for a deleted pod.
```

Comment:
s/pod/Pod

also, not sure what "handles Pod annotations" mean in the context of deletion